carina-webdriver/src/main/java/com/qaprosoft/carina/core/foundation/webdriver/decorator/ExtendedWebElement.java

```diff
@@ -864,9 +875,27 @@ public boolean isElementPresent(long timeout) {
 		if (element != null) {
 			waitCondition = ExpectedConditions.and(ExpectedConditions.visibilityOf(element),
 					ExpectedConditions.presenceOfElementLocated(getBy()));
+			boolean tmpResult = waitUntil(waitCondition, 0);
+
+			if (tmpResult) {
+				return true;
+			}
+
+			if (originalException != null && StaleElementReferenceException.class.equals(originalException.getClass())) {
+				LOGGER.debug("StaleElementReferenceException detected in isElementPresent!");
+				try {
+					refindElement();
+					waitCondition = ExpectedConditions.and(ExpectedConditions.visibilityOf(element),
+							ExpectedConditions.presenceOfElementLocated(getBy()));
+				} catch (NoSuchElementException e) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://52.53.207.202/sonarqube/coding_rules#rule_key=squid%3AS1166)